app/src/stores/tag.ts

```diff
@@ -32,19 +61,105 @@ export const useTagStore = defineStore("category", () => {
 
     /**
      * Get tags by tag type. Only return tags that have posts.
-     * TODO: Change to return tags which are used in posts and/or other tags
      */
     const tagsByTagType = computed(() => {
-        return (tagType: TagType) => {
-            if (postStore) {
-                return tags.value?.filter((t) => {
-                    const posts = postStore.postsByTag(t._id, {});
-                    return t.tagType == tagType && posts && posts.length > 0;
-                });
-            }
-            return [];
+        return (tagType: TagType, queryOptions?: tagQueryOptions) => {
+            // Get pinned tags
+            const res = tagQuery(tags, tagType, queryOptions, "pinned");
+
+            // Get unpinned tags
+            return res.concat(tagQuery(tags, tagType, queryOptions, "unpinned"));
         };
     });
 
     return { tag, tags, tagsByTagType };
 });
+
+function tagQuery(
+    tags: Readonly<Ref<Tag[] | undefined>>,
+    tagType: TagType,
+    queryOptions?: tagQueryOptions,
+    pinned?: "pinned" | "unpinned" | "any",
+) {
+    const postStore = usePostStore();
+    if (postStore) {
```

Comment:
This if is not necessary, `usePostStore` will always return either an existing store or create it. It was needed before because the store was created at the top of the file, which is not the right place.